src/CollaboraMediaPermissions.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Drupal\collabora_online;
+
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\Entity\BundlePermissionHandlerTrait;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\media\MediaTypeInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides permissions for Collabora per media type.
+ *
+ * This class is registered in collabora_online.permissions.yml.
+ *
+ * @see \Drupal\media\MediaPermissions
+ */
+class CollaboraMediaPermissions implements ContainerInjectionInterface {
+  use BundlePermissionHandlerTrait;
+  use StringTranslationTrait;
+
+  /**
+   * The entity type manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * MediaPermissions constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager service.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager) {
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static($container->get('entity_type.manager'));
+  }
+
+  /**
+   * Returns an array of media type permissions.
+   *
+   * @return array
+   *   The media type permissions.
+   *
+   * @see \Drupal\user\PermissionHandlerInterface::getPermissions()
+   */
+  public function mediaTypePermissions(): array {
+    // Generate media permissions for all media types.
+    $media_types = $this->entityTypeManager->getStorage('media_type')->loadMultiple();
+    return $this->generatePermissions($media_types, [$this, 'buildPermissions']);
+  }
+
+  /**
+   * Returns a list of permissions for a given media type.
+   *
+   * @param \Drupal\media\MediaTypeInterface $type
+   *   The media type.
+   *
+   * @return array
+   *   An associative array of permission names and descriptions.
+   */
+  protected function buildPermissions(MediaTypeInterface $type) {
+    $type_id = $type->id();
+    $type_params = ['%type_name' => $type->label()];
+
+    return [
+      "preview $type_id in collabora" => [
+        'title' => $this->t('%type_name: Preview media file in Collabora', $type_params),
+      ],
+      "edit own $type_id in collabora" => [
+        'title' => $this->t('%type_name: Edit own media file in Collabora', $type_params),
+      ],
+      "edit any $type_id in collabora" => [
```

Comment:
We had internal spec which said "edit all", but "edit any" is more aligned with core permissions.

Reply:
Also, "edit any xyz" works better with singular "xyz". With "edit all xyz", xyz would have to be plura.